keyboards/avalanche/avalanche.c

```diff
@@ -0,0 +1,4 @@
+// Copyright 2022 Vitaly Volkov (@vlkv)
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+#include "avalanche.h"
```

Comment:
This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "avalanche.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    } else if (index == 1) { /* Second encoder */
        if (clockwise) {
            tap_code(KC_PGUP);
        } else {
            tap_code(KC_PGDOWN);
        }
    }
    return true;
}
#endif

```

Reply:
Thanks, I was wondering about a way to share code like this but was only able to find the userspace feature. Is there something similar for sharing oled screen behavior?

Reply:
> Is there something similar for sharing oled screen behavior?

Yup!  `oled_task_kb`, and return false on `!oled_task_user()`. Eg

```c
bool oled_task_kb(void) {
    if (!oled_task_user()) {
        return false;
    }
    // run keyboard code
}
```